equinix/fabric_connection_schema.go

```diff
@@ -0,0 +1,840 @@
+			Computed:    true,
+			Description: "Unique Resource URL",
+		},
+	}
+}
+
+func createServiceProfileSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Service Profile URI response attribute",
+		},
+		"type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice([]string{"L2_PROFILE", "L3_PROFILE", "ECIA_PROFILE", "ECMC_PROFILE"}, true),
+			Description:  "Service profile type",
+		},
+		"name": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Customer-assigned service profile name",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix assigned service profile identifier",
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "User-provided service description",
+		},
+	}
+}
+
+var createAccessPointLinkProtocolSchRes = &schema.Resource{
+	Schema: createAccessPointLinkProtocolSch(),
+}
+
+func createAccessPointLinkProtocolSch() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Description:  "Type of the link protocol",
+			ValidateFunc: validation.StringInSlice([]string{"UNTAGGED", "DOT1Q", "QINQ", "EVPN_VXLAN"}, true),
+		},
+		"vlan_tag": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Vlan Tag information, vlanTag value specified for DOT1Q connections",
+		},
+		"vlan_s_tag": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Vlan Provider Tag information, vlanSTag value specified for QINQ connections",
+		},
+		"vlan_c_tag": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Vlan Customer Tag information, vlanCTag value specified for QINQ connections",
+		},
+		"vni": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "vni",
+		},
+	}
+}
+
+var createAccessPointVirtualDeviceRes = &schema.Resource{
+	Schema: createAccessPointVirtualDeviceSch(),
+}
+
+func createAccessPointVirtualDeviceSch() map[string]*schema.Schema {
+
+	return map[string]*schema.Schema{
+		"href": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Unique Resource Identifier",
+		},
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix-assigned Virtual Device identifier",
+		},
+		"type": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Virtual Device type",
+		},
+	}
+}
+
+func createAccessPointInterface() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"uuid": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Equinix-assigned interface identifier",
+		},
+		"id": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "id",
```

Comment:
I suspect `id` should not be here.

Reply:
Same element we discussed in read data source, it is not independent resource.

Reply:
nested part of main object and is not created, updated, deleted independently.